internals/secrethub/run.go

```diff
@@ -400,31 +400,68 @@ func (e EnvFile) Secrets() []string {
 // NewEnv loads an environment of key-value pairs from a string.
 // The format of the string can be `key: value` or `key=value` pairs.
 func NewEnv(raw string, vars map[string]string) (EnvSource, error) {
-	templates, err := parseEnv(raw)
+	env, parser, err := parseEnvironment(raw)
 	if err != nil {
-		template, ymlErr := parseYML(raw)
-		if ymlErr != nil {
-			return nil, err
-		}
-		return template, nil
+		return nil, err
 	}
 
-	secretTemplates := make(map[string]tpl.SecretTemplate, len(templates))
-	for k, template := range templates {
+	secretTemplates := make(map[string]tpl.SecretTemplate, len(env))
+	for _, envvar := range env {
+		err = validation.ValidateEnvarName(envvar.key)
```

Comment:
This has to be `Posix` right?

Reply:
This hasn't been changed (https://github.com/secrethub/secrethub-cli/blob/v0.23.0/internals/secrethub/run.go#L346). Do we want it to?

Reply:
Hmm ok, I don't think we need to. Though not entirely sure what the impact is for the `KEY=VALUE` pattern.